custom_components/localtuya/cover.py

```diff
@@ -190,6 +190,8 @@ def status_restored(self, stored_state):
 
     def status_updated(self):
         """Device status was updated."""
+        super.status_updated(self)
```

Comment:
This is the buggy line. The `status_updated` method expects the `status `parameter, which is not being passed.

Reply:
Aha. Very sorry about missing that one. I unfortunately don't have any covers to test with. There's actually a status_updated() function that doesn't take a status parameter...  However, I made 2 mistakes in the above line. It should've been super().status_updated().

I'll fix and push the change through.

Reply:
Could you re-test the change with your covers?

Reply:
Looks like it is working now. Let me check the stability for a couple of days and then I'll merge it.
You may want to have it tested by people owning different devices such as fans, climates, lights to check that no other issues arise with the other platforms, since the modification are very extended.

Reply:
No, something is going wrong. The covers receive commands and can move but their position is not updated (I use them in Timed mode). I need you to fix this before I can merge it.

Reply:
Yep - you're right. The super.status_updated definition isn't compatible with the code that comes after that call in the cover.status_updated definition. I've reworked that function slightly, so that _should_ address the issue. Unfortunately i don't have any Tuya cover devices, so i can't re-test. When you have a chance can you test and let me know if that's fixed it?

Reply:
OK now everything seems to be working back as normal. I'll probably merge this between today and tomorrow.
Please keep ready to provide support and fixes in case other platforms will show issues related to this PR, due to the extension of the new code.
Thank you for contributing to the project!

Reply:
No worries at all - thank you for maintaining localtuya as a project :)

I'll keep an eye out for bugs, but feel free to tag me in any you feel are related in case I miss something.